prov/rxm/src/rxm_cq.c

```diff
@@ -491,20 +505,24 @@ ssize_t rxm_cq_handle_large_data(struct rxm_rx_buf *rx_buf)
 				rx_buf, rx_buf->recv_entry->total_len);
 		}
 		total_recv_len -= copy_len;
+
+		RXM_DEC_TX_CREDITS(rx_buf->ep, ret);
+		if (OFI_UNLIKELY(ret)) {
+			ret = rxm_rndv_defer_readv(i, iov, desc, count, rx_buf);
+			if (OFI_UNLIKELY(ret))
+				goto readv_err;
+		}
```

Comment:
If the send queue is processed by MSG provider / HW (verbs) at a faster rate than the application reads completions, we get a completion queue overflow. We won't get EAGAIN for send because the HW is processing it faster. The credits are needed to avoid overflowing completion queue. I'll update the commit message.